realm/realm-library/src/androidTest/kotlin/io/realm/mixed/MixedParameterizedQueryTests.kt

```diff
@@ -0,0 +1,964 @@
+/*
+ * Copyright 2021 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.mixed
+
+import androidx.test.platform.app.InstrumentationRegistry
+import io.realm.*
+import io.realm.entities.*
+import io.realm.kotlin.where
+import org.bson.types.Decimal128
+import org.bson.types.ObjectId
+import org.junit.*
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+import java.lang.IllegalStateException
+import java.util.*
+import kotlin.reflect.KFunction
+import kotlin.reflect.jvm.javaMethod
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+class ParameterizedTest(
+        val function: KFunction<*>,
+        val parameters: Array<Any?>,
+        val results: Array<Any?>
+) {
+    @Suppress("UNCHECKED_CAST")
+    private fun asMixed(array: Array<Any?>): Array<Mixed> {
+        val mixedArray = arrayOfNulls<Mixed?>(array.size)
```

Comment:
Nitpick: you could try to write this in a more Kotlin-esque way by using `array.map`. Same goes for the method `generateMixedValues` above. Not required, just would make it look more functional.

Reply:
Does using array.map throw for undefined elements? I like to catch when we miss to define a Mixed type (future-proofing)